internal/services/storage/storage_share_directory_resource.go

```diff
@@ -269,7 +304,11 @@ func storageShareDirectoryRefreshFunc(ctx context.Context, client *directories.C
 	return func() (interface{}, string, error) {
 		res, err := client.Get(ctx, id.ShareName, id.DirectoryPath)
 		if err != nil {
-			return nil, strconv.Itoa(res.HttpResponse.StatusCode), fmt.Errorf("retrieving %s: %v", id, err)
+			state := "unknown"
+			if res.HttpResponse != nil {
+				state = strconv.Itoa(res.HttpResponse.StatusCode)
+			}
+			return nil, state, fmt.Errorf("retrieving %s: %v", id, err)
```

Comment:
Given that the statechangeconf is expecting `404` errors as "Pending" this is still bugged?
The error check above should possibly be:
```go 
	if err != nil && !response.WasNotFound(res.HttpResponse) {
```